posts/indu-plp-part3/plp-part3.qmd

```diff
@@ -0,0 +1,147 @@
+- **Preprocessing**: included handling missing values, one-hot encoding, normalization, and creating train-test splits.
+- **Model training**: used the `MLJ.jl` framework to train logistic regression, random forest, and XGBoost models.
+- **Evaluation**: involved computing AUC and accuracy for binary classification.
+
+## Reflections on the OHDSI Framework vs Julia Approach
+
+The OHDSI ecosystem, particularly its **Patient Level Prediction** package in R, offers an end-to-end solution that integrates tightly with ATLAS and standardized vocabularies. It handles cohort creation, covariate extraction, modeling, and visualization all in one environment.
+
+<br>
+<center>
+  ![](./ohdsi_plp.png)
+
+  OHDSI Patient Level Prediction Package in R
+</center>
+<br>
+
+In contrast, building this pipeline in Julia provided greater flexibility and control. Each step, from cohort extraction to model evaluation, had to be explicitly implemented. While this required more effort, it allowed deeper insight into how features were generated, how cohorts were structured, and how model performance was affected by upstream choices.
+
+## Model performance
+
+After preprocessing the features and labels, model performance was not as strong as expected. The AUC values for the classifiers were relatively low for the models:
+
+- Logistic Regression (L1-regularized)
+- Random Forest: AUC 
+- XGBoost: AUC
+
+These unexpectedly low model performances prompted a deeper investigation into the root causes. The primary issue was that the data available in our synthetic OMOP CDM was not well-suited to answer the research question: “From those diagnosed with hypertension, who goes on to develop diabetes?” 
+
+Temporal context was also missing, as the features extracted were basic binary indicators that did not reflect the timing or frequency of clinical events. The one-year prediction window might have been too narrow for diabetes to develop meaningfully after a hypertension diagnosis. 
+
+Additionally, many patients had very limited observational data before the index date, reducing the reliability of the features. Altogether, the problem lay not with the modeling itself but with the mismatch between the available data and the clinical nuances of the question being asked.
+
+## Lessons Learned
+
+This project highlighted that Julia is well-suited for observational health research workflows. Its strong data ecosystem (`DataFrames.jl`, `DuckDB.jl`, `CSV.jl`), rich machine learning interfaces (`MLJ.jl`), and increasing support from domain-specific packages like `OHDSICohortExpressions.jl` make it a compelling environment for building transparent, customizable pipelines. Unlike black-box tools, Julia allows full control over every stage-from data access to model tuning-encouraging deeper understanding and reproducibility.
+
+From this experience, several lessons became evident:
```

Comment:
Can you add more of your own perspectives here about your own pain points? No worries about making it super concise or clean with AI at the moment; just use your own voice here.